### PR TITLE
Staticstic Collector refactor

### PR DESCRIPTION
**Pull request by eladc-git**
## Pull Request Description:
1. refactor:
- move "quantization_analyzer" content to StatisitcCollector init
- remove "attach_sc" per framework implementation (don't need it here)
3. add logs for some scenarios
4. change mean collector from IIR to standard mean
5. change histogram merge bin width to fixed interval: 1/n_bins

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

**Comment by ofirgo** [comment]
Can you specify in the PR description what is the refactor? what is the change in the computation that you made?
(also remember to include it in the commit message once we merge the PR)

File: model_compression_toolkit/core/common/model_collector.py [hunk 1]
```diff
@@ -17,67 +17,129 @@
...
         self.fw_info = fw_info
 
-        node2fetch = []  # List of graph nodes, the model should output their outputs.
-        stats_containers_list = []  # List of output statistics containers of nodes ordered
-        # the same as node2fetch so statistics of outputs can be gathered for the correct statistics container.
-
-        for n in self.graph.nodes():
-            out_stats_container = self.graph.get_out_stats_collector(n)
+        # Assign statisitcs collectors to nodes
+        nodes_sorted = topological_sort(graph)
```
Comment (ofirgo) [inline model_compression_toolkit/core/common/model_collector.py:101]: can you use `graph.get_topo_sorted_nodes()`?

File: model_compression_toolkit/core/common/quantization/quantization_analyzer.py [hunk 2]
```diff
@@ -1,71 +0,0 @@
...
-
-    """
-    current_tensor = graph.get_out_stats_collector(node)
-    is_list_nostat_collectors = isinstance(current_tensor, list) and len([sc for sc in current_tensor if not isinstance(sc, common.NoStatsCollector)]) == 0
-    if isinstance(current_tensor, common.NoStatsCollector) or current_tensor is None or is_list_nostat_collectors:
-        out_channel_axis = fw_info.out_channel_axis_mapping.get(node.type)
-        graph.set_out_stats_collector_to_node(node, common.StatsCollector(out_channel_axis))
-
-
-def analyzer_graph(node_analyze_func: Callable,
```
Comment (ofirgo) [inline model_compression_toolkit/core/common/quantization/quantization_analyzer.py:41]: why did you removed this?

**Review by ofirgo — CHANGES_REQUESTED** [review]
weiting for answers regarding some minor points

File: model_compression_toolkit/core/common/quantization/quantization_analyzer.py (hunk 2 above)
Comment (eladc-git) [reply model_compression_toolkit/core/common/quantization/quantization_analyzer.py:41]: This "analyer" only sets statistic collectors to nodes. I moved it from "quantization" to model collector.

**Review by ofirgo — APPROVED** [review]

**Pull request merged**